lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLTranslatorUtils.java

```diff
@@ -1366,49 +1366,65 @@ public static List<ColumnModel> getSchemaOfSelect(SelectList selectList,
 	 * @return
 	 */
 	public static ColumnModel getSchemaOfDerivedColumn(DerivedColumn derivedColumn,
-			TableAndColumnMapper tableAndColumnMapper) {
-		// the SelectColumn provides a starting name and type.
+			TableAndColumnMapper tableAndColumnMapper) 
+	{
+		// The SelectColumn provides a starting name and type.
 		SelectColumn selectColumn = getSelectColumns(derivedColumn, tableAndColumnMapper);
-		Long maximumSize = null;
-		Long maxListLength = null;
-		List<String> enumValues = null;
 		// The data type is correctly inferred by the #getSelectColumns call
 		ColumnType columnType = selectColumn.getColumnType();
-		String defaultValue = null;
-		FacetType facetType = null;
-		List<JsonSubColumnModel> jsonSubColumns = null;
+		
+		// Local object to store temporary column attributes while we build them
+		class ColumnAttributes {
+	        Long maximumSize = null;
+	        Long maxListLength = null;
+	        String defaultValue = null;
+	        FacetType facetType = null;
+	        List<String> enumValues = null;
+	        List<JsonSubColumnModel> jsonSubColumns = null;
+	    }
+		final ColumnAttributes attrs = new ColumnAttributes();
+		
+		// If we have an existing selectColumn id, we can model the columnModel off of its attributes
 		if(selectColumn.getId() != null) {
 			ColumnModel cm = tableAndColumnMapper.getColumnModel(selectColumn.getId());
-			maximumSize = cm.getMaximumSize();
-			maxListLength = cm.getMaximumListLength();
-			defaultValue = cm.getDefaultValue();
-			facetType = cm.getFacetType();
-			enumValues = cm.getEnumValues();
-			jsonSubColumns = cm.getJsonSubColumns();
-		}else {
-			for (ColumnReference cr : derivedColumn.createIterable(ColumnReference.class)) {
-				ColumnTranslationReference ctr = tableAndColumnMapper.lookupColumnReference(cr).orElse(null);
-				if (ctr != null) {
-					maximumSize = addLongsWithNull(maximumSize, ctr.getMaximumSize());
-					maxListLength = addLongsWithNull(maxListLength, ctr.getMaximumListLength());
-					defaultValue = ctr.getDefaultValues();
-					facetType = ctr.getFacetType();
-					jsonSubColumns = ctr.getJsonSubColumns();
-				}
-			}
+			attrs.maximumSize = cm.getMaximumSize();
+			attrs.maxListLength = cm.getMaximumListLength();
+			attrs.defaultValue = cm.getDefaultValue();
+			attrs.facetType = cm.getFacetType();
+			attrs.enumValues = cm.getEnumValues();
+			attrs.jsonSubColumns = cm.getJsonSubColumns();
+		} else {
+			derivedColumn.createIterable(ValueExpressionPrimary.class).forEach(vep -> {
+				switch(vep.getChild().getClass().getName()) {
+				
+					case "org.sagebionetworks.table.query.model.UnsignedValueSpecification":

```

Comment:
In the version of Java we are using the switch only works with a few basic types.  It looks like they will be improving this in Java 17+.  Until then we need to work with what we have.  The problem with matching on the string of a class name, is that this will break if the class is moved or renamed.
Also a switch with only one case plus a default is identical to a simple if/else statement.
However, in this case you should have three possible matches on the child:
1. UnsignedValueSpecification
2. ColumnReference
3. Any other case
For the second case you do not need to call vep.getFirstElementOfType() because the child already be a ColumnReference.
Here is a structure I would accept:
if(vep.getChild() instanceof UnsignedValueSpecification){
...
} else if (vep.getChild() instanceof ColumnReference){
    tableAndColumnMapper.lookupColumnReference((ColumnReference)vep.getChild())...
}